### PR TITLE
add service and changes profile model

### PR DESCRIPTION
**Pull request by MiryamGabay**

### CODE REVIEW

**Review by avishag-gvili — COMMENTED** [review]
הוספת service-profile

File: server-side/models/profile.model.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
 });
 
+// Middleware to check the status of listWebsites before saving
+profileSchema.pre('save', function (next) {
+    const profile = this;
 
+    if (profile.statusBlockedSites === 'white list') {
+        profile.listWebsites.forEach(site => {
+            if (site.status === 'block') {
+                return next(new Error('In white list mode, site status cannot be "block"'));
```
Comment (SariGlick) [inline server-side/models/profile.model.js:27]: add suitable status code

File: server-side/models/profile.model.js [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
     }]
 });
 
+// Middleware to check the status of listWebsites before saving
+profileSchema.pre('save', function (next) {
+    const profile = this;
 
+    if (profile.statusBlockedSites === 'white list') {
+        profile.listWebsites.forEach(site => {
+            if (site.status === 'block') {
```
Comment (SariGlick) [inline server-side/models/profile.model.js:26]: missing check for open & black list

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by avishag-gvili — COMMENTED** [review]
הוספת profile-service

**Pull request merged**